vm/devices/vmbus/vmbus_server/src/lib.rs

```diff
@@ -843,6 +864,16 @@ impl ServerTask {
             self.server
                 .with_notifier(&mut self.inner)
                 .open_complete(offer_id, status);
+
+            // Some guests ignore interrupts before they receive the OpenResult message. To avoid
+            // a potential hang, signal the channel after a delay if needed.
+            if let Some(delay) = self.channel_unstick_delay {
+                let mut timer = PolledTimer::new(&self.driver);
+                self.channel_unstickers.push(Box::pin(async move {
```

Comment:
Seems like the guest could cause unbounded memory use by closing and opening a channel very rapidly.